modules/library/metadata/src/test/java/org/geotools/util/URLsTest.java

```diff
@@ -112,6 +112,7 @@ public void testUrlToFile() throws Exception {
         URL url = new URL("file", "////oehhwsfs09", "/some/path/on/the/server/filename.nds");
         File windowsShareFile = URLs.urlToFile(url);
         assertNotNull(windowsShareFile);
+        assertEquals("file café", URLs.urlToFile(new URL("file:file%20caf%C3%A9")).toString());
```

Comment:
Test case is a bit sparse, but adequate. For completeness, testing all of the following might be better, bit it is OK as-is:

`file:file%20caf%C3%A9"`
`file:/file%20caf%C3%A9"`
`file://file%20caf%C3%A9"`

Reply:
@johngschulz please add the extra tests suggested by @tbarsballe. We will thank you in the future.

Reply:
@bencaradocdavies I've added the additional test, when converting urlToFile using `file:/` the single slash remains in place.  I am unsure if this is an expected behavior or not.